### PR TITLE
Use setAttribute, not properties to fix SVG update

### LINKED ISSUE #278: HTML + Binding
I've tried to build `HTML` which will react to external events from external components(such as Button click), through "Binding" wrapper.

struct SVGCircle View 7

It does render with intial value, but it doesn't re-render after sending new value. 

Questions: 
* How to properly handle this?
* It this a bug? 
* Should I use DynamicHTML instead?

### PR DESCRIPTION
**Pull request by MaxDesiatov**
Property assignment does not update SVG elements. We may want to use properties instead of `setAttribute` in the future for optimizations. For now I think that `setAttribute` works in all cases, including SVG, and seems safer to me.

Resolves #278.

### CODE REVIEW

**Comment by j-f1** [comment]
This causes a regression in the TextField demo — try typing in the first field then the second field on both `main` and this branch. Ultimately the solution imo is to have a list of keys that should be set as properties instead of attributes.

**Comment by MaxDesiatov** [comment]
I've introduced a new `HTMLAttribute` type to apply updates cleanly. There's no regression anymore, at least in the `TextField` example 🙂

File: Sources/TokamakStaticHTML/Views/HTML.swift [hunk 1]
```diff
@@ -17,10 +17,35 @@
...
+ */
+public struct HTMLAttribute: Hashable {
+  public let value: String
+  public let isUpdatedAsProperty: Bool
+
+  public init(_ value: String, isUpdatedAsProperty: Bool) {
+    self.value = value
+    self.isUpdatedAsProperty = isUpdatedAsProperty
+  }
+}
```
Comment (j-f1) [inline Sources/TokamakStaticHTML/Views/HTML.swift:33]: Neat idea! How do you feel about adding a helper constructor so you can just do `.property("some-name")`?

```suggestion

    static func property(_ value: String) -> HTMLAttribute {
    .init(value, isUpdatedAsProperty: true)
  }
}
```

**Comment by carson-katri** [comment]
Is the updatedAsProperty a requirement for certain attributes or certain element & attribute combos?

**Comment by MaxDesiatov** [comment]
IDK, I wasn't able to find any information on this other than some people warning that `setAttribute` could be somehow slower that property assignment, but no appropriate benchmarks were present to verify this.

**Comment by j-f1** [comment]
I think it’s usually based on specific names. Here’s React’s file that defines how it handles attributes and properties: https://github.com/facebook/react/blob/92c7e49895032885cffaad77a69d71268dda762e/packages/react-dom/src/shared/DOMProperty.js

**Comment by MaxDesiatov** [comment]
Interesting that `value` is not in this list 🤔

```javascript
// These are the few React props that we set as DOM properties
// rather than attributes. These are all booleans.
[
  'checked',
  // Note: `option.selected` is not updated if `select.multiple` is
  // disabled with `removeAttribute`. We have special logic for handling this.
  'multiple',
  'muted',
  'selected',
```

**Comment by carson-katri** [comment]
If its a set list, could we just automatically check them instead of each element specifying if it needs to be treated as a property?

**Comment by MaxDesiatov** [comment]
I thought about it, but attributes are created everywhere on every render cycle, which would mean that the check would be multiplied by the number of times you use any attribute whatsoever. I'm not sure what's the best way to avoid the check on attribute creation other than hardcoding it as a property at the place of use.

**Comment by MaxDesiatov** [comment]
Another option would be to forbid attribute creation from a string literal and predefine all attributes in `TokamakStaticHTML` with an appropriate `isUpdatedAsProperty` value.

**Comment by carson-katri** [comment]
You could do it with type-safety and allow a string literal:
```swift
struct HTMLAttribute : ExpressibleByStringLiteral {
  ...
  static let value: Self = .init("value", isUpdatedAsProperty: true)
}
```
Something like that?

**Comment by MaxDesiatov** [comment]
Yeah, it wouldn't prevent anyone from still initializing `value` attribute from a string literal though. Probably still better than the current option...

**Comment by carson-katri** [comment]
Could make it more explicit with a `.custom("myAttribute")` initializer.

**Comment by MaxDesiatov** [comment]
Oh, requiring `.custom` is going to require changing hundreds of lines, I'd rather do that in a separate PR 😅

**Comment by j-f1** [comment]
it looks like there are a bunch of files in [`react-dom/src/client`](https://github.com/facebook/react/tree/92c7e49895032885cffaad77a69d71268dda762e/packages/react-dom/src/client) that handle form controls individually (`ReactDOMInput.js`, `ReactDOMSelect.js`, `ReactDOMOption.js`, `ReactDOMTextarea.js`) which seem to have the value handler inside of them.

**Review by j-f1 — APPROVED** [review]
Either way I think this is a good start to unblock our user, and we can iterate more later.

**Review by carson-katri — APPROVED** [review]

**Pull request merged**